cpp/source/deal.II/geometry.templates.h

```diff
@@ -60,31 +60,154 @@ void check_no_overlap(std::unordered_map<std::string, std::set<T>> &map)
 }
 
 template <int dim>
+void build_general_cell(std::vector<dealii::Point<dim>> const vertices,
+                        dealii::distributed::Triangulation<dim> &tria)
+{
+  BOOST_ASSERT_MSG(vertices.size() == std::pow(2, dim),
+                   "Wrong number of vertices.");
+  // First create an hyper_cube and then deform it.
+  dealii::GridGenerator::hyper_cube(tria);
+
+  typename dealii::Triangulation<dim>::active_cell_iterator cell =
+      tria.begin_active();
+  for (unsigned int i = 0; i < dealii::GeometryInfo<dim>::vertices_per_cell;
+       ++i)
+  {
+    dealii::Point<dim> &v = cell->vertex(i);
+    v = vertices[i];
+  }
+}
+
+void build_trapezoid_1(dealii::Point<2> const &dimensions,
+                       dealii::distributed::Triangulation<2> &tria)
+{
+  std::vector<dealii::Point<2>> vertices(4);
+  double const edge = 1e-5;
```

Comment:
You should consider moving edge outside of the `build_trapezoid_{1,2,3}` routines.